EntranceRando.py

```diff
@@ -40,17 +40,21 @@ def __init__(self, rng: random.Random):
     # todo - investigate whether this might leak memory (holds references to Entrances)?
     @staticmethod
     @functools.cache
-    def _is_dead_end(entrance: Entrance):
+    def _is_dead_end(entrance: Entrance, visited_regions=""):
```

Comment:
Doing cycle detection correctly here is a bit nuanced, probably best to just leave it out for now and I'll figure out how to deal with it after merging

The main issue with current implementation is
1. Visited regions should be a set, and then you can check `if region in visited_regions`
2. Regardless of the type, this arg will mess with the caching behavior